protocol/communities/manager.go

```diff
@@ -5134,3 +5134,54 @@ func (m *Manager) GetCuratedCommunities() (*CuratedCommunities, error) {
 func (m *Manager) SetCuratedCommunities(communities *CuratedCommunities) error {
 	return m.persistence.SetCuratedCommunities(communities)
 }
+
+func getImage(image *images.IdentityImage) (*common.LinkPreviewThumbnail, error) {
+	thumbnail := &common.LinkPreviewThumbnail{}
+
+	if image.IsEmpty() {
+		return nil, nil
+	}
+
+	width, height, err := images.GetImageDimensions(image.Payload)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get image dimensions: %w", err)
+	}
+
+	dataURI, err := image.GetDataURI()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get data uri: %w", err)
+	}
+
+	thumbnail.Width = width
+	thumbnail.Height = height
+	thumbnail.DataURI = dataURI
+	return thumbnail, nil
+}
+
+func (m *Manager) BuildStatusCommunityLinkPreview(community *Community) (*common.StatusCommunityLinkPreview, error) {
```

Comment:
Thanks for this!
We should also reuse this in `linkpreview_unfurler_status.go`, otherwise it's still a duplication.

_____

Also, do you think we could make it a method of `Community` type?
It would make much more sense than as a method of Manager I think.
```go
func (community *Community) toStatusLinkPreview() (*common.StatusCommunityLinkPreview, error)
```

... and same for images:
```go
func (image *images.IdentityImage) toLinkPreveiwThumbnail() (*common.LinkPreviewThumbnail, error)
```

Or make it the other way around a method `StatusCommunityLinkPreview` and `LinkPreviewThumbnail` correspondingly.

_______

If you want, you can leave this as is and I will refactor it after this PR.

Reply:
Did for the community one, But fell for a circular dependency when doing the image one.